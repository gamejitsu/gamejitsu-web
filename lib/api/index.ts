import * as t from "io-ts"
import { ModelType } from "../schemas"
import { Model, Schemas, Attrs, Attr, Relationship, Embedded } from "../schema"

export type ResponseType = "one" | "many"

export type AttributesC<T extends ModelType> = t.TypeC<
  {
    [K in keyof Schemas[T]["_T"]]: Schemas[T][K] extends Attr | Embedded
      ? AttributeC<Schemas[T][K]>
      : never
  }
>

export type NonNullableRelationshipsC<T extends ModelType = ModelType> = t.TypeC<
  {
    [K in keyof Schemas[T]["_T"]]: Schemas[T][K] extends Relationship
      ? RelationshipC<Schemas[T][K]>
      : never
  }
>

export type RelationshipsC<T extends ModelType> = t.UnionC<
  [NonNullableRelationshipsC<T>, t.UndefinedC]
>

export type AttributeC<T extends Attr | Embedded> = T extends Attr
  ? T["isOptional"] extends true
    ? Attrs[T["type"]]
    : t.UnionC<[Attrs[T["type"]], t.UndefinedC]>
  : T extends Embedded
  ? T["modelType"]
  : never

type RelationshipDataC<T extends Relationship> = t.TypeC<{
  type: t.LiteralC<T["modelType"]>
  id: t.StringC
}>

export type RelationshipC<T extends Relationship = Relationship> = t.TypeC<{
  data: T["type"] extends "one" ? RelationshipDataC<T> : t.ArrayC<RelationshipDataC<T>>
}>

export type ModelC<T extends ModelType = ModelType> = t.TypeC<{
  type: t.LiteralC<T>
  id: t.StringC
  attributes: AttributesC<T>
  relationships: RelationshipsC<T>
}>

export type DataC<T extends ModelType, U extends ResponseType> = U extends "one"
  ? ModelC<T>
  : t.ArrayC<ModelC<T>>

export type IncludedC = t.UnionC<[t.ArrayC<ModelC<ModelType>>, t.UndefinedC]>

export type DeserializedData<T extends ModelType, U extends ResponseType> = U extends "one"
  ? Model<T>
  : Model<T>[]

export type DeserializedIncluded = Record<ModelType, Model[] | undefined>

export interface DeserializedResponse<T extends ModelType, U extends ResponseType> {
  data: DeserializedData<T, U>
  included: DeserializedIncluded
}

export * from "./api"
export { deserializeResponse } from "./response"
