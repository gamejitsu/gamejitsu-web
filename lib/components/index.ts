import Button from "./Button"
import ButtonNew from "./ButtonNew"
import ButtonIcon from "./ButtonIcon"
import ButtonAlternative from "./ButtonAlternative"
import ButtonDark from "./ButtonDark"
import Card from "./Card"
import CardNew from "./CardNew"
import Form from "./Form"
import FormGroup from "./FormGroup"
import HeroImage from "./HeroImage"
import HeroImageSmall from "./HeroImageSmall"
import ImageButton from "./ImageButton"
import InputGroup from "./InputGroup"
import Layout from "./Layout"
import LayoutWithMenu from "./LayoutWithMenu"
import LayoutWithMenuUser from "./LayoutWithMenuUser"
import Link from "./Link"
import LinkLeftMenu from "./LinkLeftMenu"
import Navbar from "./Navbar"
import NavbarLeftMenu from "./NavbarLeftMenu"
import SkillLevelSelect from "./SkillLevelSelect"
import Spinner from "./Spinner"
import SteamButton from "./SteamButton"
import Title from "./Title"

export {
  Button,
  ButtonNew,
  ButtonIcon,
  ButtonAlternative,
  ButtonDark,
  Card,
  CardNew,
  Form,
  FormGroup,
  HeroImage,
  HeroImageSmall,
  ImageButton,
  InputGroup,
  Layout,
  LayoutWithMenu,
  LayoutWithMenuUser,
  Link,
  LinkLeftMenu,
  Navbar,
  NavbarLeftMenu,
  SkillLevelSelect,
  Spinner,
  SteamButton,
  Title
}
