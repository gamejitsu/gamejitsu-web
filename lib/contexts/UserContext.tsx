import { createContext } from "react"
import { User } from "gamejitsu/api/resources/user"

const UserContext = createContext<User | null>(null)

export default UserContext
