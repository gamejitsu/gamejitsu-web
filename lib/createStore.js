import { createStore as reduxCreateStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import reducer from './reducer'
import saga from './saga'

const createStore = initialState => {
  const sagaMiddleware = createSagaMiddleware()
  const store = reduxCreateStore(reducer, initialState, applyMiddleware(sagaMiddleware))
  store.sagaTask = sagaMiddleware.run(saga)
  return store
}

export default createStore
