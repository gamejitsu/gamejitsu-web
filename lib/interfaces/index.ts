import { FunctionComponent } from "react"

export interface AuthenticatedComponent extends FunctionComponent {
  skipAuthentication?: boolean
}

export interface Model {
  id: string
}
