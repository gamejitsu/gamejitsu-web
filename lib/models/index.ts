import { Model } from "../api/response"

export type ReviewRequest = Model<"review-request">
export type Session = Model<"session">
export type User = Model<"user">
export type Replay = Model<"replay">
export type Review = Model<"review">
