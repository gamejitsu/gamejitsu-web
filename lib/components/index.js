import Button from './Button'
import ButtonSteam from './ButtonSteam'
import Card from './Card'
import Comment from './Comment'
import Layout from './Layout'
import Link from './Link'
import Navbar from './Navbar'
import Spinner from './Spinner'
import UserContext from './UserContext'

export { Button, ButtonSteam, Card, Layout, Link, Navbar, Spinner, UserContext, Comment}
