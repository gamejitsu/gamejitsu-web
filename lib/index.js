import createStore from './createStore'
import theme from './theme'

export { createStore, theme }
