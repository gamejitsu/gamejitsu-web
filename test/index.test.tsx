import { mount } from "enzyme"
import React from "react"
import { ThemeProvider } from "styled-components"
import { theme } from "gamejitsu"
import DemoPage from "../pages/demo"
import { UserContext } from "../lib/contexts"

const user = {
  type: "user" as "user",
  username: "userTest",
  isSyncingReplays: false,
  steamId: "1",
  hasPublicProfile: true,
  email: "foobar@example.com",
  coachId: "10328f42-849e-405d-affc-15373340w0bw",
  id: "10328f42-849e-405d-affc-3537w340cwbf"
}

test("Index page should match snapshot", () => {
  const home = mount(
    <ThemeProvider theme={theme}>
      <UserContext.Provider value={user}>
        <DemoPage />
      </UserContext.Provider>
    </ThemeProvider>
  )

  expect(home).toMatchSnapshot()
})
