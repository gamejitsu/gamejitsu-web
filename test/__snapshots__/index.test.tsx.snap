// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "activeColor": "#ffD74f",
      "backgroundColor": "#111",
      "errorColor": "#ff665a",
      "lightBackgroundColor": "#252526",
      "linkColor": "#fff",
      "primaryColor": "#05F26C",
      "secondaryColor": "#3518E5",
      "textColor": "#F1F2E9",
      "textFont": "'Exo 2',sans-serif",
    }
  }
>
  <Page>
    <Layout>
      <Navbar>
        <Navbar__Container
          alignItems="center"
          color="white"
          px={2}
          py={2}
        >
          <StyledComponent
            alignItems="center"
            color="white"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": [Function],
                "__emotion_real": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    Object {
                      "boxSizing": "border-box",
                      "margin": 0,
                      "minWidth": 0,
                    },
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    Object {
                      "display": "flex",
                    },
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                },
                "__emotion_styles": Array [
                  Object {
                    "boxSizing": "border-box",
                    "margin": 0,
                    "minWidth": 0,
                  },
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  Object {
                    "display": "flex",
                  },
                ],
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Navbar__Container-sc-1qe3s5x-0",
                  "isStatic": false,
                  "lastClassName": "iNtICZ",
                  "rules": Array [
                    "background-color:",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "Navbar__Container",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Navbar__Container-sc-1qe3s5x-0",
                "target": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    Object {
                      "boxSizing": "border-box",
                      "margin": 0,
                      "minWidth": 0,
                    },
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    Object {
                      "display": "flex",
                    },
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                },
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            px={2}
            py={2}
          >
            <Styled(div)
              alignItems="center"
              className="Navbar__Container-sc-1qe3s5x-0 iNtICZ"
              color="white"
              px={2}
              py={2}
            >
              <div
                className="Navbar__Container-sc-1qe3s5x-0 iNtICZ css-p1imuz"
              >
                <NavLink
                  href="/"
                >
                  <Styled(div)
                    m={1}
                  >
                    <div
                      className="css-1ee7t57"
                    >
                      <Link
                        href="/"
                      >
                        <Link
                          href="/"
                        >
                          <Link__Content
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Link__Content-j0sqir-0",
                                    "isStatic": false,
                                    "lastClassName": "jgHxZN",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";cursor:pointer;",
                                    ],
                                  },
                                  "displayName": "Link__Content",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Link__Content-j0sqir-0",
                                  "target": "a",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={[Function]}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="Link__Content-j0sqir-0 jgHxZN"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                Gamejitsu
                              </a>
                            </StyledComponent>
                          </Link__Content>
                        </Link>
                      </Link>
                    </div>
                  </Styled(div)>
                </NavLink>
                <NavLink
                  href="/dashboard"
                >
                  <Styled(div)
                    m={1}
                  >
                    <div
                      className="css-1ee7t57"
                    >
                      <Link
                        href="/dashboard"
                      >
                        <Link
                          href="/dashboard"
                        >
                          <Link__Content
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Link__Content-j0sqir-0",
                                    "isStatic": false,
                                    "lastClassName": "jgHxZN",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";cursor:pointer;",
                                    ],
                                  },
                                  "displayName": "Link__Content",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Link__Content-j0sqir-0",
                                  "target": "a",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={[Function]}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="Link__Content-j0sqir-0 jgHxZN"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                Dashboard
                              </a>
                            </StyledComponent>
                          </Link__Content>
                        </Link>
                      </Link>
                    </div>
                  </Styled(div)>
                </NavLink>
                <NavLink
                  href="/reviews"
                >
                  <Styled(div)
                    m={1}
                  >
                    <div
                      className="css-1ee7t57"
                    >
                      <Link
                        href="/reviews"
                      >
                        <Link
                          href="/reviews"
                        >
                          <Link__Content
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Link__Content-j0sqir-0",
                                    "isStatic": false,
                                    "lastClassName": "jgHxZN",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";cursor:pointer;",
                                    ],
                                  },
                                  "displayName": "Link__Content",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Link__Content-j0sqir-0",
                                  "target": "a",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={[Function]}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="Link__Content-j0sqir-0 jgHxZN"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                Reviews
                              </a>
                            </StyledComponent>
                          </Link__Content>
                        </Link>
                      </Link>
                    </div>
                  </Styled(div)>
                </NavLink>
                <NavLink
                  href="/coach-reviews"
                >
                  <Styled(div)
                    m={1}
                  >
                    <div
                      className="css-1ee7t57"
                    >
                      <Link
                        href="/coach-reviews"
                      >
                        <Link
                          href="/coach-reviews"
                        >
                          <Link__Content
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Link__Content-j0sqir-0",
                                    "isStatic": false,
                                    "lastClassName": "jgHxZN",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";cursor:pointer;",
                                    ],
                                  },
                                  "displayName": "Link__Content",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Link__Content-j0sqir-0",
                                  "target": "a",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={[Function]}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="Link__Content-j0sqir-0 jgHxZN"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                Coach Reviews
                              </a>
                            </StyledComponent>
                          </Link__Content>
                        </Link>
                      </Link>
                    </div>
                  </Styled(div)>
                </NavLink>
                <Styled(div)
                  mx="auto"
                >
                  <div
                    className="css-1ntkiwo"
                  />
                </Styled(div)>
                <NavLink
                  href="/"
                  key="username"
                >
                  <Styled(div)
                    m={1}
                  >
                    <div
                      className="css-1ee7t57"
                    >
                      <Link
                        href="/"
                      >
                        <Link
                          href="/"
                        >
                          <Link__Content
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Link__Content-j0sqir-0",
                                    "isStatic": false,
                                    "lastClassName": "jgHxZN",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";cursor:pointer;",
                                    ],
                                  },
                                  "displayName": "Link__Content",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Link__Content-j0sqir-0",
                                  "target": "a",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={[Function]}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="Link__Content-j0sqir-0 jgHxZN"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                userTest
                              </a>
                            </StyledComponent>
                          </Link__Content>
                        </Link>
                      </Link>
                    </div>
                  </Styled(div)>
                </NavLink>
                <Button
                  key="logout"
                  onClick={[Function]}
                  text="Logout"
                >
                  <Button__Content
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <StyledComponent
                      disabled={false}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Button__Content-sc-1qhaj03-0",
                            "isStatic": false,
                            "lastClassName": "iUEPVJ",
                            "rules": Array [
                              "background:linear-gradient( to bottom,",
                              [Function],
                              ",",
                              [Function],
                              " );padding:12px 30px;font-size:15px;outline:0;border:0;cursor:pointer;",
                            ],
                          },
                          "displayName": "Button__Content",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Button__Content-sc-1qhaj03-0",
                          "target": "button",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      onClick={[Function]}
                      type="button"
                    >
                      <button
                        className="Button__Content-sc-1qhaj03-0 iUEPVJ"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        Logout
                      </button>
                    </StyledComponent>
                  </Button__Content>
                </Button>
              </div>
            </Styled(div)>
          </StyledComponent>
        </Navbar__Container>
      </Navbar>
      <Layout__Container
        mx="auto"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "div",
              "__emotion_forwardProp": [Function],
              "__emotion_real": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": [Function],
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  Object {
                    "boxSizing": "border-box",
                    "margin": 0,
                    "minWidth": 0,
                  },
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "defaultProps": undefined,
                "displayName": "Styled(div)",
                "render": [Function],
                "withComponent": [Function],
              },
              "__emotion_styles": Array [
                Object {
                  "boxSizing": "border-box",
                  "margin": 0,
                  "minWidth": 0,
                },
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "Layout__Container-sc-1iu1ypv-0",
                "isStatic": false,
                "lastClassName": "kXwVwM",
                "rules": Array [
                  "max-width:1024px;",
                ],
              },
              "displayName": "Layout__Container",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "Layout__Container-sc-1iu1ypv-0",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": [Function],
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  Object {
                    "boxSizing": "border-box",
                    "margin": 0,
                    "minWidth": 0,
                  },
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "defaultProps": undefined,
                "displayName": "Styled(div)",
                "render": [Function],
                "withComponent": [Function],
              },
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          mx="auto"
        >
          <Styled(div)
            className="Layout__Container-sc-1iu1ypv-0 kXwVwM"
            mx="auto"
          >
            <div
              className="Layout__Container-sc-1iu1ypv-0 kXwVwM css-1ntkiwo"
            >
              <Head>
                <Component
                  handleStateChange={null}
                  inAmpMode={false}
                  reduceComponentsToState={[Function]}
                />
              </Head>
            </div>
          </Styled(div)>
        </StyledComponent>
      </Layout__Container>
    </Layout>
  </Page>
</ThemeProvider>
`;
